Dingo/app/src/main/java/com/example/dingo/trips/TripScreen.kt

```diff
@@ -322,30 +347,144 @@ fun LocationPermissionScreen() {
+
+
+
 
     val cameraPositionState = rememberCameraPositionState {
-        position = CameraPosition.fromLatLngZoom(points.lastOrNull() ?: LatLng(51.52061810406676, -0.12635325270312533), 15f)
+        position = CameraPosition.fromLatLngZoom(trip.locations.lastOrNull() ?: LatLng(51.52061810406676, -0.12635325270312533), 15f)
     }
 
     GoogleMap(
         modifier = Modifier.fillMaxSize(),
         cameraPositionState = cameraPositionState
     ) {
-        if (points.lastOrNull() != null) {
-            Polyline(points = points)
+        if (trip.locations.lastOrNull() != null) {
+            // Convert the timestamp to a Java Date object
+            val startDate: Date = trip.startTime.toDate()
+            val endDate: Date = trip.endTime.toDate()
+
+            // Convert the timestamp to a Java Date object
+            // Create a SimpleDateFormat object with your desired date format
+            val dateFormat = SimpleDateFormat("dd-MM-yyyy HH:mm:ss")
+
+            // Format the date as desired
+
+            val formattedStartTime = dateFormat.format(startDate)
+            val formattedEndTime = dateFormat.format(endDate)
+
+
+
+            Polyline(points = trip.locations)
             Marker(
-                state = MarkerState(position = points.last()),
+                state = MarkerState(position = trip.locations.last()),
                 title = "Start",
-                snippet = "Trip Started at"
+                snippet = "Trip Started at ${formattedStartTime}"
             )
             Marker(
-                state = MarkerState(position = points.first()),
+                state = MarkerState(position = trip.locations.first()),
                 title = "Finish",
-                snippet = "You are here"
+                snippet = "Trip Ended at ${formattedEndTime}"
             )
+
+            // Assuming you have a list of picture paths and picture locations
+            val picturePaths: List<String> = trip.picturePaths
+            val pictureLocations: List<LatLng> = trip.pictureLocations
+
+//            val icon = bitmapDescriptorFromVector(
+//                LocalContext.current, R.drawable.pin
+//            )
+            // Iterate through the picture paths and picture locations
+            picturePaths.forEachIndexed { index, picturePath ->
+                val pictureLocation = pictureLocations[index]
+                // You can customize the marker icon, if desired
+//                Marker(
+//                    state = MarkerState(position = pictureLocation),
+//                    title = "Picture $index",
+//                    snippet = "Location: ${pictureLocation.latitude}, ${pictureLocation.longitude}",
+//                    icon = BitmapDescriptorFactory.fromResource(R.drawable.fauna_placeholder)
+//                )
+                val storageRef = FirebaseStorage.getInstance().reference.child(picturePath)
+                var bitmap: Bitmap? by remember {mutableStateOf(null)}

```

Comment:
Here is how you load an image from Firebase as a bitmap to load into Google maps marker